coinaddrng/validation.py

```diff
@@ -726,3 +727,106 @@ def prefixtodec(prefix):
         total += prefix[-i]*multiplier
         multiplier *= 256
     return total+prefix[-1]
+
+
+@attr.s(frozen=True, slots=True, auto_attribs=True)
+class SS58Address:
+    format: int
+    length: int
+
+
+@attr.s(frozen=True, slots=True, cmp=False)
+@implementer(IValidator)
+class SS58Validator(ValidatorBase):
+
+    name = 'SS58Check'
+    valid_ss58_format = None
+
+    def validate(self):
+        try:
+            self._ss58_decode(self.request.address, valid_ss58_format=self.valid_ss58_format)
+        except ValueError:
+            return False
+        else:
+            return True
+
+        return False
+
+    @staticmethod
+    def _decode_ss58_address_format(address: bytes) -> SS58Address:
+        if address[0] & 0b0100_0000:
+            format_length = 2
+            ss58_format = ((address[0] & 0b0011_1111) << 2) | (address[1] >> 6) | \
+                          ((address[1] & 0b0011_1111) << 8)
+        else:
+            format_length = 1
+            ss58_format = address[0]
+
+        return SS58Address(ss58_format, format_length)
```

Comment:
Tohle je malickost, ale jelikoz je to trida s 2 `int` atributy, takze bych to pro prehlednost vracel takhle:

```
return SS58Address(format=ss58_format, length=format_length)
```